### PR TITLE
ECM-585

### PR DESCRIPTION
**Pull request by hsjhita**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ECM-585

### Change description ###

Added validation to check the start time and finish time

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java [hunk 1]
```diff
@@ -87,4 +91,32 @@ public static void updatePostponedDate(CaseData caseData) {
         }
 
     }
+
+    public static List<String> hearingTimeValidation(CaseData caseData) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java:95]: Too many identations. You could simplify this

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java [hunk 2]
```diff
@@ -87,4 +91,32 @@ public static void updatePostponedDate(CaseData caseData) {
...
+            }
+        }
+        return errors;
+    }
+
+    private static void checkStartFinishTimes(List<String> errors, DateListedType dateListedType,
+                                              String hearingNumber) {
+        var startDate = LocalDateTime.parse(dateListedType.getHearingTimingStart());
+        var finishTime = LocalDateTime.parse(dateListedType.getHearingTimingFinish());
+        if (startDate.isAfter(finishTime) || startDate.isEqual(finishTime)) {
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java:118]: if (!startDate.isBefore(finishTime)) {
maybe?

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java (hunk 1 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java:95]: Refactored

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java (hunk 2 above)
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/HearingsHelper.java]: changed to if (finishTime.isBefore(startTime)
in terms of the ticket, i think it makes more sense if it's in that order

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java [hunk 3]
```diff
@@ -737,6 +737,9 @@
 
         var caseDetails = ccdRequest.getCaseDetails();
         List<String> errors = HearingsHelper.hearingMidEventValidation(caseDetails.getCaseData());
+        if ("updateHearing".equals(ccdRequest.getEventId())) {
+            errors.addAll(HearingsHelper.hearingTimeValidation(caseDetails.getCaseData()));
+        }
         //        errors.addAll(eventValidationService.validateHearingDatesNotInFuture(caseDetails.getCaseData()));
```
[the remark this answers is not in the document]
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java:743]: I'm working on that in another branch

[the remark this answers is not in the document]
Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java:743]: That validation will be refactored to work alongside this

**Review by btensay — APPROVED** [review]

**Pull request merged**